### DISCUSSION TITLE
Connecting milvus host to aws instance ip

### DISCUSSION THREAD

**Discussion by Sushanti99** [milvus-io/milvus#4717] [Q&A and General discussion] [answered]
Can somebody here help me with how to connect Milvus(host="...", port='19530') to a milvus running on docker on an ec2 instance? I can't seem to find anything in the documentation either

**Comment by stephenleo** [comment, marked as answer]
I run it on GCP Compute Engine VM without any issue. Just expose port 19530 using -p and use the internal IP address for `host` in `Milvus(host="...", port='19530')`.

P.S. I'm accessing from another VM within the same GCP project

```
sudo docker run -d --name milvus_cpu_0.11 \
-p 19530:19530 \
-p 19121:19121 \
-v /home/$USER/milvus/db:/var/lib/milvus/db \
-v /home/$USER/milvus/conf:/var/lib/milvus/conf \
-v /home/$USER/milvus/logs:/var/lib/milvus/logs \
-v /home/$USER/milvus/wal:/var/lib/milvus/wal \
milvusdb/milvus:0.11.0-cpu-d101620-4c44c0
```

**Reply by Sushanti99** [reply]
I tried to use the internal ip for host but it didn't connect to the server

**Reply by stephenleo** [reply]
To connect from a local system, the way I do it is by setting up ssh port forwarding. Something like below for GCP. (Not sure if this is the best way! But it works)

```
gcloud compute ssh --zone <your-VM-zone> <username>@<vm-internal-ip> --project <your-project> -- -L 19530:localhost:19530
```

After this, you can run `Milvus(host='localhost', port='19530')` on your local

`-L` is for ssh port forwarding [Link](https://www.ssh.com/ssh/tunneling/example)

**Reply by stephenleo** [reply]
I think this is the AWS equivalent: https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-ssh-tunnel-local.html

**Reply by Sushanti99** [reply]
Got this to work. Thank you so much for the quick response

**Reply by stephenleo** [reply]
If the answer worked, pls help to "select as answer" so that the question shows up as answered.
